tests/ui/impl_trait_in_params.rs

```diff
@@ -32,4 +35,11 @@ impl S {
     pub fn k<K: AnotherTrait<u32>>(_: K, _: impl AnotherTrait<u32>) {} //~ ERROR: `impl Trait` used as a function parameter
 }
 
+// Trying with imported traits
+impl ToString for S {
+    fn to_string(&self) -> String {
+        String::from("a")
+    }
+}
```

Comment:
This lint warns for impl trait in parameters, so I think we also need to check for a case where parameters include impl trait.

Reply:
Oh what a brain melt. I really didn't notice that 🐱